src/shogun/modelselection/GradientModelSelection.cpp

```diff
@@ -173,8 +173,7 @@ float64_t CGradientModelSelection::get_cost(SGVector<float64_t> model_vars, SGVe
 
 	// evaluate the machine
 	CEvaluationResult* evaluation_result=m_machine_eval->evaluate();
-	CGradientResult* gradient_result=CGradientResult::obtain_from_generic(
-			evaluation_result);
+	CGradientResult* gradient_result = evaluation_result->as<CGradientResult>();
```

Comment:
here we go: the previous function did a `SG_REF`, but now it is missing -> problems